### PR TITLE
Improving the RGCN_HETERO example

### PR DESCRIPTION
**Pull request by drivanov**
## Description

In the current PR we eliminate the following warnings::
```
/usr/local/lib/python3.10/dist-packages/dgl/dataloading/dataloader.py:1076: DGLWarning: Dataloader CPU affinity opt is not enabled, consider switching it on (see enable_cpu_affinity() or CPU best practices for DGL [https://docs.dgl.ai/tutorials/cpu/cpu_best_practises.html])

/usr/local/lib/python3.10/dist-packages/numpy/lib/function_base.py:495: RuntimeWarning: Mean of empty slice.
  avg = a.mean(axis)

/usr/local/lib/python3.10/dist-packages/numpy/core/_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
```
and a few `nan` outputs:
```
Epoch 00000 | Valid Acc: 0.9643 | Valid loss: 0.8942 | Time: nan
```
Newly implemented parameters `--num_workers` option specifies the number of processes that generate packages in parallel.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: examples/pytorch/rgcn-hetero/entity_classify_mb.py [hunk 1]
```diff
@@ -134,53 +141,64 @@ def main(args):
...
+                )
+                print(
+                    "Epoch {:05d} | Batch {:03d} | Train Acc: {:.4f} | Train Loss: {:.4f} | Time: {:.4f}".format(
+                        epoch,
+                        i,
+                        train_acc,
+                        loss.item(),
+                        time.time() - batch_tic,
+                    )
                 )
```
Comment (frozenbugs) [inline examples/pytorch/rgcn-hetero/entity_classify_mb.py:179]:
```suggestion
                print(
                    f"Epoch {epoch:05d} | Batch {i:03d} | Train Acc: "
                     "{train_acc:.4f} | Train Loss: {loss.item():.4f} | Time: "
                     "{time.time() - batch_tic:.4f}"
                )
```

File: examples/pytorch/rgcn-hetero/entity_classify_mb.py [hunk 2]
```diff
@@ -134,53 +141,64 @@ def main(args):
...
-        val_loss, val_acc = evaluate(
-            model, val_loader, node_embed, labels, category, device
-        )
-        print(
-            "Epoch {:05d} | Valid Acc: {:.4f} | Valid loss: {:.4f} | Time: {:.4f}".format(
-                epoch, val_acc, val_loss, np.average(dur)
+            val_loss, val_acc = evaluate(
+                model, val_loader, node_embed, labels, category, device
+            )
+            print(
```
Comment (frozenbugs) [inline examples/pytorch/rgcn-hetero/entity_classify_mb.py:187]: ditto: use f string.

File: examples/pytorch/rgcn-hetero/entity_classify_mb.py [hunk 3]
```diff
@@ -86,6 +87,12 @@ def main(args):
         labels = labels.to(device)
         embed_layer = embed_layer.to(device)
 
+    if args.num_workers <= 0:
```
Comment (frozenbugs) [inline examples/pytorch/rgcn-hetero/entity_classify_mb.py:90]: why not support args.num_workers = 0 which is existing behavior?

Comment (drivanov) [reply examples/pytorch/rgcn-hetero/entity_classify_mb.py:90]: In fact, when the `dataloader` is created with `args.num_workers=0`:
```
    dataloader = dgl.dataloading.DataLoader(
       . . .
        num_workers=args.num_workers,
    )
```
and 
```
with dataloader.enable_cpu_affinity():
```
is used, we get the following exception:
```
  File "/usr/local/lib/python3.10/dist-packages/dgl/dataloading/dataloader.py", line 1118, in enable_cpu_affinity
    raise Exception(
Exception: ERROR: affinity should be used with at least one DL worker
```

File: examples/pytorch/rgcn-hetero/entity_classify_mb.py (hunk 2 above)
Comment (drivanov) [reply examples/pytorch/rgcn-hetero/entity_classify_mb.py:187]: DONE

File: examples/pytorch/rgcn-hetero/entity_classify_mb.py (hunk 1 above)
Comment (drivanov) [reply examples/pytorch/rgcn-hetero/entity_classify_mb.py:179]: DONE

**Comment by stupidoge** [comment]
@drivanov thanks for your effort! However, for R-GCN, we cannot add any nodes' features into the model. When we use our own dataset, it is too troublesome to change the original message_func and apply_func.

could you give any convenient ways? Or could you add this new function?

thanks and best regards.

**Pull request merged**